From: Thomas Renninger <trenn@suse.de>
Subject: ACPI: Store valid ACPI tables passed via early initrd in reserved memblock areas
References: none
Patch-Mainline: Queued in subsystem tree (see Git-Repo:)
Git-commit: 53aac44c904abbad9f474f652f099de13b5c3563
Git-repo: git://git.kernel.org/pub/scm/linux/kernel/git/tip/tip

Signed-off-by: Thomas Renninger <trenn@suse.de>

A later patch will compare them with ACPI tables that get loaded at boot or
runtime and if criteria match, a stored one is loaded.

Signed-off-by: Thomas Renninger <trenn@suse.de>
Link: http://lkml.kernel.org/r/1349043837-22659-4-git-send-email-trenn@suse.de
Cc: Len Brown <lenb@kernel.org>
Cc: Robert Moore <robert.moore@intel.com>
Cc: Yinghai Lu <yinghai@kernel.org>
Cc: Eric Piel <eric.piel@tremplin-utc.net>
Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>

Automatically created from "patches.arch/003-acpi_initrd_override_tables.patch" by xen-port-patches.py

--- head.orig/arch/x86/kernel/setup-xen.c	2012-10-31 11:57:31.000000000 +0100
+++ head/arch/x86/kernel/setup-xen.c	2012-10-31 12:08:09.000000000 +0100
@@ -1082,6 +1082,8 @@ void __init setup_arch(char **cmdline_p)
 
 	reserve_initrd();
 
+	acpi_initrd_override((void *)initrd_start, initrd_end - initrd_start);
+
 #ifndef CONFIG_XEN
 	reserve_crashkernel();
 
